pkg/cmd/experimental/syncgroups/groupsyncer_test.go

```diff
@@ -1,22 +1,115 @@
 	userapi "github.com/openshift/origin/pkg/user/api"
 )
 
+func TestMakeOpenShiftGroup(t *testing.T) {
+	syncer := &LDAPGroupSyncer{
+		Out:  ioutil.Discard,
+		Err:  ioutil.Discard,
+		Host: "test-host",
+		GroupNameMapper: &TestGroupNameMapper{
+			NameMapping: map[string]string{
+				"alfa": "zulu",
+			},
+		},
+	}
+
+	tcs := map[string]struct {
+		ldapGroupUID   string
+		usernames      []string
+		startingGroups []runtime.Object
+		expectedGroup  *userapi.Group
+		expectedErr    string
+	}{
+		"bad ldapGroupUID": {
+			ldapGroupUID: "bravo",
+			expectedErr:  "no name found for group: bravo",
+		},
+		"good": {
+			ldapGroupUID: "alfa",
+			usernames:    []string{"valerie"},
+			expectedGroup: &userapi.Group{ObjectMeta: kapi.ObjectMeta{Name: "zulu",
+				Annotations: map[string]string{ldaputil.LDAPURLAnnotation: "test-host", ldaputil.LDAPUIDAnnotation: "alfa"}},
+				Users: []string{"valerie"}},
+		},
+		"replaced good": {
+			ldapGroupUID: "alfa",
+			usernames:    []string{"valerie"},
+			expectedGroup: &userapi.Group{ObjectMeta: kapi.ObjectMeta{Name: "zulu",
+				Annotations: map[string]string{ldaputil.LDAPURLAnnotation: "test-host", ldaputil.LDAPUIDAnnotation: "alfa"}},
+				Users: []string{"valerie"}},
+			startingGroups: []runtime.Object{
+				&userapi.Group{ObjectMeta: kapi.ObjectMeta{Name: "zulu",
+					Annotations: map[string]string{ldaputil.LDAPURLAnnotation: "test-host", ldaputil.LDAPUIDAnnotation: "alfa"}},
+					Users: []string{"other-user"}},
+			},
+		},
+		"conflicting uid": {
+			ldapGroupUID: "alfa",
+			usernames:    []string{"valerie"},
+			startingGroups: []runtime.Object{
+				&userapi.Group{ObjectMeta: kapi.ObjectMeta{Name: "zulu",
+					Annotations: map[string]string{ldaputil.LDAPURLAnnotation: "test-host", ldaputil.LDAPUIDAnnotation: "bravo"}},
+					Users: []string{"other-user"}},
+			},
+			expectedErr: `group "zulu": openshift.io/ldap.uid annotation did not match LDAP UID: wanted alfa, got bravo`,
+		},
+		"conflicting host": {
+			ldapGroupUID: "alfa",
+			usernames:    []string{"valerie"},
+			startingGroups: []runtime.Object{
+				&userapi.Group{ObjectMeta: kapi.ObjectMeta{Name: "zulu",
+					Annotations: map[string]string{ldaputil.LDAPURLAnnotation: "bad-host", ldaputil.LDAPUIDAnnotation: "alfa"}},
+					Users: []string{"other-user"}},
+			},
+			expectedErr: `group "zulu": openshift.io/ldap.url annotation did not match sync host: wanted test-host, got bad-host`,
+		},
+	}
+
+	for name, tc := range tcs {
+		fakeClient := testclient.NewSimpleFake(tc.startingGroups...)
+		syncer.GroupClient = fakeClient.Groups()
+
+		actualGroup, err := syncer.makeOpenShiftGroup(tc.ldapGroupUID, tc.usernames)
+		if err != nil && len(tc.expectedErr) == 0 {
+			t.Errorf("%s: unexpected error %v", name, err)
+
+		} else if err == nil && len(tc.expectedErr) != 0 {
+			t.Errorf("%s: expected %v, got nil", name, tc.expectedErr)
+
+		} else if err != nil {
+			if e, a := tc.expectedErr, err.Error(); e != a {
```

Comment:
expected, actual, testCase? Naming here makes it hard to follow for what that's worth.